playground/src/components/state-machines-list-connector/state-machines-list-connector.js

```diff
@@ -1,19 +1,25 @@
 import React from 'react';
 import PropTypes from 'prop-types';
-import { Sdk } from '@commercetools-frontend/sdk';
+import { connect } from 'react-redux';
+import { createSdkGet } from '@commercetools-frontend/sdk';
+import { ApplicationStoreContext } from '../../store';
 import * as actions from './actions';
 
-class ChannelsListConnector extends React.Component {
-  static displayName = 'ChannelsListConnector';
+const SdkGet = createSdkGet({ context: ApplicationStoreContext });
```

Comment:
Thanks for the feedback! Yes, having some abstractions might be useful, we just need to find the more suitable one. For now I would not add any abstraction and try to get a feeling on how things can be improved.
Using a nested store is not strictly necessary, if you don't provide one, the default one defined within the AppShell will be used. Therefore, people can start "adopting" the nested provider whenever and however they like. If we then add an abstraction, it should be easy to "upgrade" as it's something opt-in.

---

As for the suggestions you made, I think having static methods would not really work, because we need to differentiate between context of AppShell and nested context.

However, maybe we can invert the contexts, so the AppShell uses a custom one and apps can use the default redux store context. Let me look into that...

Reply:
> However, maybe we can invert the contexts, so the AppShell uses a custom one and apps can use the default redux store context. Let me look into that...

This won't work, mostly because of the notifications and the action creators. If we were to put that into a custom store, each "bound" notification action creator would need to connect to the specific store context, which would make the upgrade a huge pain in the ass.

On a related note, if dispatching notifications is much more straight-forward (see #68), we can internally manage that without leaking the implementation to the consumer.

I would say that for now we keep it as is. The nested provider is **not a requirement**, so apps do not need to do that for now. Maybe when we do #68, we can refactor the internals of the AppShell to use a different store context without requiring apps to do that. So better to not promote much the fact that the apps can use a nested provider.

I'll refactor the playground to keep it without, to avoid confusion. It was anyway for testing purposes to make sure things are still working.

Reply:
Done, see https://github.com/commercetools/merchant-center-application-kit/pull/197/commits/bc35bc583f9311184f961931445d16477fa51bff